arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1790,12 +1809,14 @@ int ClusterInfo::createCollectionCoordinator(std::string const& databaseName,
     CONDITION_LOCKER(locker, agencyCallback->_cv);
 
     while (true) {
-      errorMsg = *errMsg;
 
-      if (*dbServerResult >= 0) {
+      int tmpRes = dbServerResult->load(std::memory_order_acquire);
+      if (tmpRes >= 0) {
+        cbGuard.fire(); // unregister cb before accessing errMsg
+        errorMsg = *errMsg;
```

Comment:
Doesn't this still allow unprotected access to `errorMsg`/`errMsg`?

Reply:
_errMsg_ should only be written once _dbServerResult_ is set. I unregister the cb before reading _errMsg_ so the callback thread should not longer access it